components/tag-list/tag-list.js

```diff
@@ -346,13 +346,10 @@ class TagList extends LocalizeCoreElement(InteractiveMixin(ArrowKeysMixin(LitEle
 
 	_handleItemDeleted(e) {
 		const rootTarget = e.composedPath()[0];
-		const children = this._getVisibleEffectiveChildren();
-		const itemIndex = children.indexOf(rootTarget);
-
-		if (children.length <= 1) return;
+		const itemIndex = this._items.slice().indexOf(rootTarget);
 
 		setTimeout(() => {
-			const focusableElem = children[itemIndex - 1] || (children[itemIndex] === e.target ? children[itemIndex + 1] : children[itemIndex]);
+			const focusableElem = this._items[itemIndex - 1] || this._items[0] || (this._items[itemIndex] === e.target ? this._items[itemIndex + 1] : this._items[itemIndex]);
```

Comment:
This logic looks a bit different than it was previously. Is there a different case that's trying to tackle?